pkg/ipcache/metadata.go

```diff
@@ -43,3 +64,343 @@ func GetIDMetadataByIP(prefix string) labels.Labels {
 	defer idMDMU.RUnlock()
 	return IdentityMetadata[prefix]
 }
+
+// InjectLabels injects labels into the IdentityMetadata (IDMD) map. The given
```

Comment:
Yes, the comment is wrong here. Good spotting.

From reviewing the actual code, I think the code is accurate to the diagram but this comment is intended to say something like:

```
// InjectLabels injects labels from the IdentityMetadata (IDMD) map into the IPCache. The given  
// source is the source of the caller, as inserting into the IPCache requires    
// where this updated information is coming from.
```

Reply:
(It's likely that this comment is a bit older before I pushed back on having this piece of logic manipulate IDMD, that likely explains how we ended up here; furthermore I have a tendency to review the code in more detail than the comments since the comments can be misleading but the code does what the code says)

Reply:
> // InjectLabels injects labels from the IdentityMetadata (IDMD) map into the IPCache. The given  
// source is the source of the caller, as inserting into the IPCache requires    
// where this updated information is coming from.

Looks good to me :+1: 

> furthermore I have a tendency to review the code in more detail than the comments since the comments can be misleading but the code does what the code says

:heavy_plus_sign: except when I not familiar with the codebase and need help from comments to understand. So I guess it's good that my ipcache-newbie eyes read that code after all :smile: